pkg/deploy/controller/deployment/controller.go

```diff
@@ -289,6 +290,14 @@ func (c *DeploymentController) makeDeployerPod(deployment *kapi.ReplicationContr
 
 	gracePeriod := int64(10)
 
+	// In case the deployment config strategy does not set the resource requests,
+	// we default them.
+	if deploymentConfig.Spec.Strategy.Resources.Limits == nil {
+		deploymentConfig.Spec.Strategy.Resources.Limits = kapi.ResourceList{}
+		deploymentConfig.Spec.Strategy.Resources.Limits[kapi.ResourceCPU] = resource.MustParse("100m")
+		deploymentConfig.Spec.Strategy.Resources.Limits[kapi.ResourceMemory] = resource.MustParse("256Mi")
```

Comment:
@smarterclayton - we go limits to requests... i am not sure going from requests to limits will give you the desired behavior otherwise, you will want to set these requests REALLY small.